### PR TITLE
23.x backports

### PR DESCRIPTION
**Pull request by jonatack**
Backport the following to 23.x:

- #24468
- #24528
- #24527
- #24609
- #24555 
- #24663
- #24572 
- #24636 
- #24553 
- #24659
- #24521 
- #24523
- #24690

### CODE REVIEW

**Comment by fanquake** [comment]
These are missing the `Github-Pull` and `Rebased-From` metadata in the commit messages.

**Comment by jonatack** [comment]
> These are missing the `Github-Pull` and `Rebased-From` metadata in the commit messages.

Thanks @fanquake, done.

**Comment by jonatack** [comment]
Updated along with the PR description with the merged pulls.

**Comment by fanquake** [comment]
> with the merged pulls.

Are you going to add the rest of the merged pulls that need backporting?

**Comment by jonatack** [comment]
Closing as these aren't considered as needing to be in v23.

File: src/rpc/blockchain.cpp [hunk 1]
```diff
@@ -1481,7 +1481,7 @@ static void SoftForkDescPushBack(const CBlockIndex* blockindex, UniValue& softfo
     // BIP9 status
     bip9.pushKV("status", get_state_name(current_state));
     bip9.pushKV("since", g_versionbitscache.StateSinceHeight(blockindex->pprev, consensusParams, id));
-    bip9.pushKV("status-next", get_state_name(next_state));
+    bip9.pushKV("status_next", get_state_name(next_state));
```
Comment (MarcoFalke) [inline src/rpc/blockchain.cpp:1484]: If we don't backport this, it should be reverted in master again

File: src/chainparams.cpp [hunk 2]
```diff
@@ -397,7 +397,7 @@ class CRegTestParams : public CChainParams {
         consensus.BIP65Height = 1;  // Always active unless overridden
         consensus.BIP66Height = 1;  // Always active unless overridden
         consensus.CSVHeight = 1;    // Always active unless overridden
-        consensus.SegwitHeight = 1; // Always active unless overridden
+        consensus.SegwitHeight = 0; // Always active unless overridden
```
Comment (MarcoFalke) [inline src/chainparams.cpp:400]: If we don't backport this, it should be reverted in master again

**Review by maflcko — COMMENTED** [review]
Then they should be reverted in master

**Comment by jonatack** [comment]
I think they should all be backported. But I don't decide.

File: src/rpc/blockchain.cpp (hunk 1 above)
Comment (jonatack) [reply src/rpc/blockchain.cpp:1484]: IMO it should be backported along with the other pulls here, but it's not up to me.

**Comment by jonatack** [comment]
Re-opened per IRC feedback.

**Comment by hebasto** [comment]
> #24523 would be a good backport candidate too, if it happens to be merged before the next release candidate

It is ready for backporting now.

**Comment by jonatack** [comment]
> > #24523 would be a good backport candidate too, if it happens to be merged before the next release candidate
> 
> It is ready for backporting now.

Done but there was a merge conflict in `configure.ac`. I took all the changes from that branch; don't hesitate to have a look.

File: configure.ac [hunk 3]
```diff
@@ -1438,13 +1438,21 @@ if test "$use_external_signer" != "no"; then
     ;;
     *)
       AC_MSG_CHECKING([whether Boost.Process can be used])
+      TEMP_CXXFLAGS="$CXXFLAGS"
+      dnl Boost 1.78 requires the following workaround.
+      dnl See: https://github.com/boostorg/process/issues/235
+      CXXFLAGS="$CXXFLAGS -Wno-error=narrowing"
+      TEMP_CPPFLAGS="$CPPFLAGS"
+      CPPFLAGS="$CPPFLAGS $BOOST_CPPFLAGS"
```
Comment (fanquake) [inline configure.ac:1446]: There are changes in 5dbed641eb118ab3f72a93f2b84dfc0220609f93 that are not part of the cherry-picked commit (532c64a7264dd3c7329e8839547837c57da7dbe8).

Comment (jonatack) [reply configure.ac:1446]: Done. Those were from the merge conflict. Updated with the following diff to make the commits the same:
```diff
diff --git a/configure.ac b/configure.ac
index a2f425ce93..a37553de8e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1458,8 +1458,6 @@ if test "$use_external_signer" != "no"; then
       dnl Boost 1.78 requires the following workaround.
       dnl See: https://github.com/boostorg/process/issues/235
       CXXFLAGS="$CXXFLAGS -Wno-error=narrowing"
-      TEMP_CPPFLAGS="$CPPFLAGS"
-      CPPFLAGS="$CPPFLAGS $BOOST_CPPFLAGS"
       TEMP_LDFLAGS="$LDFLAGS"
       dnl Boost 1.73 and older require the following workaround.
       LDFLAGS="$LDFLAGS $PTHREAD_CFLAGS"
@@ -1467,7 +1465,6 @@ if test "$use_external_signer" != "no"; then
         [have_boost_process="yes"],
         [have_boost_process="no"])
       LDFLAGS="$TEMP_LDFLAGS"
-      CPPFLAGS="$TEMP_CPPFLAGS"
       CXXFLAGS="$TEMP_CXXFLAGS"
       AC_MSG_RESULT([$have_boost_process])
       if test "$have_boost_process" = "yes"; then
```

**Review by hebasto — APPROVED** [review]
~ACK c5cc0e5865c6ba4475d224a78361108f81e33c8d~ Sorry, retracted.

I've backported all commits locally, and got zero diff with this PR branch.

File: configure.ac (hunk 3 above)
Comment (hebasto) [reply configure.ac]: The reason of the conflict is #24521 which has not been backported.

File: doc/README.md [hunk 4]
```diff
@@ -73,6 +73,7 @@ The Bitcoin repo's [root README](/README.md) contains relevant information on th
 - [Assets Attribution](assets-attribution.md)
 - [Assumeutxo design](assumeutxo.md)
 - [bitcoin.conf Configuration File](bitcoin-conf.md)
+- [CJDNS Support](cjdns.md)
```
Comment (hebasto) [inline doc/README.md:76]: This link is dead.

File: configure.ac (hunk 3 above)
Comment (MarcoFalke) [reply configure.ac]: Maybe it is easier to just backport @https://github.com/bitcoin/bitcoin/pull/24521 as well at this point?

File: doc/README.md (hunk 4 above)
Comment (jonatack) [reply doc/README.md:76]: Good catch. The commit adding that file is missing.

File: configure.ac (hunk 3 above)
Comment (jonatack) [reply configure.ac:1446]: > Maybe it is easier to just backport @#24521 as well at this point?

Done.

File: doc/README.md (hunk 4 above)
Comment (jonatack) [reply doc/README.md:76]: Thanks! Fixed the link at https://github.com/jonatack/bitcoin/blob/23.x-onlynet-help-and-doc-improvements/doc/README.md#miscellaneous by adding missing commits from #24555 and checked the other backports.

**Comment by jonatack** [comment]
Addressed @hebasto's catch, added #24521 per https://github.com/bitcoin/bitcoin/pull/24512#discussion_r839261495, and updated the PR description.  #24710 looks RFM and would be good to add.

**Comment by laanwj** [comment]
List-of-commits ACK 174af33d5d78cfe341b688f5393d9acc8c4aaf6b, I think we should merge this and move forward with rc3..

**Comment by hebasto** [comment]
Why #24710 is listed among backported PRs?

**Review by hebasto — APPROVED** [review]
ACK 174af33d5d78cfe341b688f5393d9acc8c4aaf6b

I've backported all commits locally without any conflicts, and got zero diff with this PR branch.

> I think we should merge this and move forward with rc3.

Agree.

**Comment by fanquake** [comment]
> I think we should merge this and move forward with rc3..

Yes. The list of commits has blown up, and this has dragged on long enough. I don't think there's a reason to wait for other documentation changes.

> Why https://github.com/bitcoin/bitcoin/pull/24710 is listed among backported PRs?

Not sure. I've removed it.

**Comment by jonatack** [comment]
> Why #24710 is listed among backported PRs?

https://github.com/bitcoin/bitcoin/pull/24512#issuecomment-1084283299

**Pull request merged**